### PR TITLE
{infra/os-runner, compute/lab}: Fix docker and add explicit prerequisites to `User-Level Threads`

### LINKED ISSUE #296: User-Level Threads/Create: cmake not found
The os-runner container does not contain `cmake` and `boost`. It can be manually installed using:
```
sudo apt update
sudo apt install cmake
sudo apt install libboost-all-dev
```

Moreover, after installing and running the commands from the lab, we get the following error:
```
 student@os:~/.../support/user-level-threads/build$ make
Scanning dependencies of target yield_barrier
[ 10%] Building CXX object CMakeFiles/yield_barrier.dir/yield_barrier.cc.o
[ 20%] Linking CXX executable yield_barrier
[ 20%] Built target yield_barrier
Scanning dependencies of target threads_and_fibers
[ 30%] Building CXX object CMakeFiles/threads_and_fibers.dir/threads_and_fibers.cc.o
[ 40%] Linking CXX executable threads_and_fibers
/usr/bin/ld: CMakeFiles/threads_and_fibers.dir/threads_and_fibers.cc.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/threads_and_fibers.dir/build.make:88: threads_and_fibers] Error 1
make[1]: *** [CMakeFiles/Makefile2:111: CMakeFiles/threads_and_fibers.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

This error can be solved by adding the following line to `CMakeLists.txt`:
```
SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

### PR DESCRIPTION
**Pull request by cristian-vijelie**
Closes #296

### CODE REVIEW

**Review by teodutu — APPROVED** [review]
Feel free to merge this if @elena19m agrees with it.

**Comment by elena19m** [comment]
I'm getting the following error when trying to install the required packages in the `os-runner container`:

```
🐳 student@os:~/.../chapters/compute/lab$ sudo apt update
...
🐳 student@os:~/.../chapters/compute/lab$ sudo apt install cmake libboost-context-dev libboost-fibers-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package libboost-fibers-dev
```

**Comment by cristian-vijelie** [comment]
Yes, sorry, it's `libboost-fiber-dev`. Solved

File: content/chapters/compute/lab/content/user-level-threads.md
Comment (elena19m) [inline content/chapters/compute/lab/content/user-level-threads.md:17]:
```suggestion
student@os:~$ sudo apt update
student@os:~$ sudo apt install cmake libboost-context-dev libboost-fiber-dev
```

Comment (elena19m) [reply content/chapters/compute/lab/content/user-level-threads.md:17]: I don't have a strong opinion regarding using `apt` instead of `apt-get`, I've seen that both versions are used in this repository. Nevertheless, `update` must be run before `install`.

However, if the docker image will be updated, these lines won't be needed.

**Review by elena19m — APPROVED** [review]
Everything works fine for me, I've approved the PR. However, the `ghcr.io/open-education-hub/os-runner:latest` image needs to be updated as well.

File: content/chapters/compute/lab/content/user-level-threads.md
Comment (gabrielmocanu) [reply content/chapters/compute/lab/content/user-level-threads.md:17]: IMHO we don't need this section since the Docker Image will be updated after this PR it's merged.

Comment (teodutu) [reply content/chapters/compute/lab/content/user-level-threads.md:17]: We should keep them, in case someone runs our content natively. @cristian-vijelie add a sentence that those commands should not be run if using the Docker image.

Comment (gabrielmocanu) [reply content/chapters/compute/lab/content/user-level-threads.md:17]: OK. 
@cristian-vijelie 
After this issue is resolved and the PR merge, just check if the `OS Runner Image` pipeline started.
You can ping me if any errors occur on that pipeline.

**Pull request merged**
